lib/services/api/session.ts

```diff
@@ -1,44 +1,87 @@
-import { HttpResponse } from "./http_service";
+import { SessionRenewalResponse } from './responses';
 
-export class Session {
+export abstract class Session {
+  public abstract canExpire(): boolean;
 
-  public accessToken: string
-  public expireAt?: number
-  public refreshToken?: string
-  public validUntil?: number
+  /** Return the token that should be included in the header of authorized network requests */
+  public abstract get authorizationValue(): string;
 
-  static FromRaw(raw: any) {
-    return new Session(raw.accessToken, raw.expireAt, raw.refreshToken, raw.validUntil);
+  static FromRawStorageValue(raw: any) {
```

Comment:
Fixed here: https://github.com/standardnotes/snjs/pull/110/commits/92565a6f4f94a46918e2f9644b575993221ecb61

I had to go with `&` instead of `|` for combining the expected types, even though I would have preferred `|`. But the compiler didn't create they were all optional properties. Let me know if you know a different way.

Reply:
Do you mean that the compiler wouldn't let you check for things like `raw.jwt`? I think you should keep it as '|' because that's what callers expect, and if you can't find a way to make the compiler behave, force its hand. It's a small function and forcing input correctness with the compiler is more important IMO.
By the way, when working with `unknown` Typescript wants you to use the `in` operator https://www.typescriptlang.org/docs/handbook/advanced-types.html